packages/x-date-pickers/src/internals/models/props/baseToolbarProps.ts

```diff
@@ -3,22 +3,21 @@ import { CalendarOrClockPickerView } from '../views';
 import type { PickerOnChangeFn } from '../../hooks/useViews';
 import type { ExportedCalendarPickerProps } from '../../../CalendarPicker/CalendarPicker';
 import type { ExportedClockPickerProps } from '../../../ClockPicker/ClockPicker';
+import { PickerStatePickerProps } from '../../hooks/usePickerState';
 
-export interface BaseToolbarProps<TDate>
+export interface BaseToolbarProps<TDate, TValue>
```

Comment:
I added a generic so that the `date` prop of BaseToolbarProps` can be correctly typed with `TValue` (the prop will be renamed `parsedValue` in #4736)